os/fs/smartfs/smartfs_utils.c

```diff
@@ -2136,6 +2136,15 @@ ssize_t smartfs_append_data(FAR struct smartfs_mountpt_s *fs, FAR struct smartfs
 				fdbg("Error writing sector %d data, ret : %d\n", readwrite.logsector, ret);
 				return ret;
 			}
+			/* If this is a newly chained data sector, write the chainheader->type for the sector */
+			if (chainheader != NULL) {
```

Comment:
Append_data is called only when it extend file, by ftruncate or O_APPEND or something.
Anyway it is only for write.
So type is always same,  this had better moved to above, before call FS_IOCTL(WRITESECT)
If it is already set type, it will skipped by physically, so it is ok.

Reply:
Yes, this is only needed when we allocate a new sector for file write.

But for first sector, type is passed from calling function (ftruncate or smartfs_write), hence I checked if (chainheader != NULL)

I think you mean that chainheader and data should both be written in one FS_IOCTL call,
Did I get that right?

Reply:
I think apply type only in sync_internal will be fine.
There we also update number of written bytes.

Reply:
For that, we will have to pass one extra parameter to sync_internal...
Is that necessary?